### PR TITLE
Add basic compatibility with RaidcoreGG-Nexus

### PR DESCRIPTION
**Pull request by Masgalor**
This PR adds the necessary basics so nexus will recognize the link-dll as a valid addon.

It keeps the existing functions to load the addon intact.
I verified that every method is still working, but did no in depth testing, I don't see any conflicts or edge-cases to test.

Nexus expects the addon to know it's own version, I've set it accordingly.
While it is recommended, it is not really necessary to update it as long as the dll is installed by hand.

Note: The GitHub-button will be hidden in this scenario.

### CODE REVIEW

File: burrito_link/dllmain.c [hunk 1]
```diff
@@ -396,3 +396,55 @@ uintptr_t mod_release() {
...
+
+extern __declspec(dllexport) struct AddonDefinition* GetAddonDef()
+{
+	AddonDef.Signature = -217;
+	AddonDef.APIVersion = 6; // taken from Nexus.h
+	AddonDef.Name = "Burrito Link";
+	AddonDef.Version.Major = 1;
+	AddonDef.Version.Minor = 4;
+	AddonDef.Version.Build = 0;
+	AddonDef.Version.Revision = 1;
```
Comment (AsherGlick) [inline burrito_link/dllmain.c:442]: I think these values should probably be dynamically populated, almost certainly  via preprocesor values passed in via cmake/make. Semver is always an annoying methodology to follow because it usually requires either manual intervention or a persisted git-adjacent datastore. Maybe we should just transition to a date-based system like many other programs use in order to avoid having to manually track this. I'll need to think about that some more. Feel free to propose suggestions.

File: burrito_link/dllmain.c [hunk 2]
```diff
@@ -396,3 +396,55 @@ uintptr_t mod_release() {
...
+
+	/* update fallback */
+	signed int      Provider;       /* What platform is the the addon hosted on */
+	const char*     UpdateLink;     /* Link to the update resource */
+
+} AddonDef;
+
+extern __declspec(dllexport) struct AddonDefinition* GetAddonDef()
+{
+	AddonDef.Signature = -217;
```
[the remark this answers is not in the document]
Comment (Masgalor) [reply burrito_link/dllmain.c:436]: The comments in the type definition say to choose a random negative number, that's what I did.
```
signed int      Signature;      /* Raidcore Addon ID, set to random unqiue negative integer if not on Raidcore */
```
Even if it says the number must be unique, there were no problems when I used an already existing number.
The important part is to use a negative number, because official Raidcore-addons use positive numbers.

File: burrito_link/dllmain.c (hunk 1 above)
Comment (Masgalor) [reply burrito_link/dllmain.c:442]: I think this depends on whether we want to work towards the advanced PR (including updates).
For arcdps we always report the version as `arc_exports.out_build = "1.0";` we could do the same here.

But in general it would be better to show the real version, alone for reporting errors and such.
I will see if can come up with a good suggestion.

File: burrito_link/dllmain.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (AsherGlick) [reply burrito_link/dllmain.c:436]: I wonder what this number's use is then if it does not need to be unique. Maybe it is a feature delta has not yet added to nexus? Or "negative number" is just a catchall for "non raidcore plugin".

[the remark this answers is not in the document]
Comment (Masgalor) [reply burrito_link/dllmain.c:436]: I did some research and joined the developers discord (which is not as useful as I thought).
The signature is used to...
 - determine if the addon is from a third party
 - identify the addon so it does not get loaded more than once
 - manage dependencies with other addons (I could not find much information about this topic)

Maybe there are more features planned, I don't know.
All addons I checked use 6 random digits, should I change the value?

[the remark this answers is not in the document]
Comment (AsherGlick) [reply burrito_link/dllmain.c:436]: Nice research

Looks like it is a regular signed integer, so we have 32 bits to play with. Lets  be cheeky here.

```
# Negative
1000 0000  0000 0000  0000 0000  0000 0000

# Burrito (ASCII)
B          r          t          o
0100 0010  0111 0010  0111 0100  0110 1111

# Negative & Burrito
1100 0010  0111 0010  0111 0100  0110 1111

# sint32
-1032686481
```

Lets go with
```
AddonDef.Signature = -1032686481;
```

[the remark this answers is not in the document]
Comment (Masgalor) [reply burrito_link/dllmain.c]: I like the way you think. :-)
Done.

File: burrito_link/dllmain.c (hunk 1 above)
Comment (Masgalor) [reply burrito_link/dllmain.c:442]: Maybe like this.

```
string(REPLACE "." ";" BURRITO_VERSION_LIST ${BURRITO_VERSION})
list(LENGTH BURRITO_VERSION_LIST BURRITO_VERSION_LIST_LENGTH)
if(BURRITO_VERSION_LIST_LENGTH EQUAL 4)
    list(GET BURRITO_VERSION_LIST0 BURRITO_VERSION_MAJOR)
    list(GET BURRITO_VERSION_LIST1 BURRITO_VERSION_MINOR)
    list(GET BURRITO_VERSION_LIST2 BURRITO_VERSION_BUILD)
    list(GET BURRITO_VERSION_LIST2 BURRITO_VERSION_REVISION)
else()
    message( FATAL_ERROR "Version format mismatch." )
endif()

add_compile_definitions(BURRITO_VERSION_MAJOR=\"${BURRITO_VERSION_MAJOR}\")
add_compile_definitions(BURRITO_VERSION_MINOR=\"${BURRITO_VERSION_MINOR}\")
add_compile_definitions(BURRITO_VERSION_BUILD=\"${BURRITO_VERSION_BUILD}\")
add_compile_definitions(BURRITO_VERSION_REVISION=\"${BURRITO_VERSION_REVISION}\")
```

Comment (Masgalor) [reply burrito_link/dllmain.c:442]: I suggest that we proceed as follows.

1. We merge this PR as is, with hardcoded values for the version.
    At the moment none of the executables or libraries we build is aware of its version, and there is no need to change this with this PR.
2. We publish a new release with the current version scheme, that would be `alpha-1.5` I think.
   There are enough changes/fixes in the `main` branch to have a new release anyway and so we have a working dll in the release-bundle. This is what matters most for our users.
3. I will open another PR where we can discuss the version scheme (I don' see whats the problem with semantic versioning and git-tags) and implement "version awareness" (don't know how else to call this) in every binary we build.
4. We can revisit the #327 to make automatic updates work if we want.

**Comment by AsherGlick** [comment]
I am mostly in agreement.
The largest problem with semantic versioning in this project is that it is not automatic, and I am loathe to add manual processes to things when I can avoid it because I am error prone and lazy human. The tag based solution has a couple of drawbacks:
* The tag needs to be in place before the release is cut
* Non-major builds dont have unique version numbers

In an ideal versioning system every commit has its own version number. One solution to this is to have the version number be the commit hash of the commit the build is built from. Commit hashes are great for uniqueness and answering the question `is my build the newest build` by comparing the `newest build hash` against `my build hash`. However, it is not so great an answering the question `is this build newer than that build`. Most of the time the only question that needs to be answered is `is my build the newest build`, so this is my preferred versioning solution in general.

Right now releases are cut entirely using the github ui, this means that the tags are added after the commit is pushed. As such there will be a CI action that runs on the commit to build it before it gets the new tag's version number. This could be mitigated a couple ways, but importantly it is something that could go wrong each release.

We could change how releases are cut, instead of manually doing anything rely entirely on pushing a tag with a version number up. If that became the release process then it would eliminate the risk of publishing a binary with the wrong version number, but I take pause at the fact that github has deprecated their first-party release cutting actions for some reason.

Another option besides hashes is the time-based solution. Not quite as good as the hash based solution as a single commit may have multiple build numbers if it is compiled twice, but does solve the issue of `is this build newer than that build` for automatically built binaries on a single branch.

That said, I think you make a valid point about this version number not really mattering right now because this PR does not address the auto-update. In that case I suggest that we change this version number to be as low of a number as possible, for two reasons:
1) It allows us the most freedom to play with the version numbers in the future
2) Seeing version `1.4.0` in nexus when a user has just downloaded `1.5.0` could cause confusion. Seeing version `0.0.0` might still be confusing, but is less confusing because it is not an actual version.

As for making a alpha-1.5 release after this PR: sounds good to me.

**Comment by Masgalor** [comment]
I changed the version to `0.0.0.1`, anything else I should do before this can be merged?
